src/category/repositories/category.repository.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { DataSource, Repository } from 'typeorm';
+
+import { ErrorCodes } from '@common/types/error';
+import { categoryMessages } from '@common/constants/messages';
+import { Category } from '../entities/category.entity';
+import { CreateCategoryDto } from '../dto/create-category.dto';
+import { UpdateCategoryDto } from '../dto/update-category.dto';
+
+@Injectable()
+export class CategoryRepository extends Repository<Category> {
+  constructor(private dataSource: DataSource) {
+    super(Category, dataSource.createEntityManager());
+  }
+
+  async createCategory(createCategoryDto: CreateCategoryDto) {
+    try {
+      const category = this.create(createCategoryDto);
+      await this.save(category);
+
+      return category;
+    } catch (error) {
+      this.handleDbError(error);
+    }
+  }
+
+  async updateCategory(id: string, updateCategoryDto: UpdateCategoryDto) {
+    const category = await this.preload({ id, ...updateCategoryDto });
+
+    if (!category)
+      throw new BadRequestException(categoryMessages.CATEGORY_NOT_FOUND);
+
+    try {
+      await this.save(category);
+      return category;
+    } catch (error) {
+      this.handleDbError(error);
+    }
+  }
+
+  private handleDbError(error: any) {
```

Comment:
Could you avoid the usage of any here?

Reply:
I'll extend the error type but I think `any` is well used here